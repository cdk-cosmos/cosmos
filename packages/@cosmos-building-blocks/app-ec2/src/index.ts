// export {

// } from './app-alb-nlb-combo';

export * from './app-alb-nlb';
