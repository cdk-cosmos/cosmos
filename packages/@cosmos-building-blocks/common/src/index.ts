export { CDKToolkit } from './cdktoolkit';
export { CrossAccountExports, CrossAccountExportsFn, CrossAccountExportsProps } from './cross-account-exports';
