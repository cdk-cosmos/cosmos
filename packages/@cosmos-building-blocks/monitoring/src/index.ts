export { ecsMetric } from './helpers';
export { ClusterDashboard, ClusterDashboardProps } from './dashboard/cluster';
