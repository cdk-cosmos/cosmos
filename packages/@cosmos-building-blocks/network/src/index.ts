export { IResolverRule, ResolverRule, ResolverRuleProps } from './route53/resolver-rule';
export {
  IResolverRuleAssociation,
  ResolverRuleAssociation,
  ResolverRuleAssociationProps,
} from './route53/resolver-rule-association';

export { ITransitGateway, TransitGateway, TransitGatewayProps } from './transit-gateway/transit-gateway';
export {
  ITransitGatewayAttachment,
  TransitGatewayAttachment,
  TransitGatewayAttachmentProps,
  TransitGatewayRoute,
  TransitGatewayRouteProps,
} from './transit-gateway/transit-gateway-attachment';

export { TransparentProxy, TransparentProxyProps } from './internet/transparent-proxy';
