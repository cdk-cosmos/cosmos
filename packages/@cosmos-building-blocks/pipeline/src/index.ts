export { AppNodePipeline, AppNodePipelineProps } from './app-node-pipeline';
