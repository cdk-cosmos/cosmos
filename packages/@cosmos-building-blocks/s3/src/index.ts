// export {

// } from './app-alb-nlb-combo';

export * from './s3';
