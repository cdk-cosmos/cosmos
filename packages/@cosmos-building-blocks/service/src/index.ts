export { EcsService, EcsServiceProps } from './ecs';
