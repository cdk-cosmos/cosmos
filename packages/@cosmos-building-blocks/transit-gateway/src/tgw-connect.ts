import { Construct } from '@aws-cdk/core';
import { IVpc, CfnRoute, CfnTransitGatewayAttachment } from '@aws-cdk/aws-ec2';
import { CfnResolverRuleAssociation } from '@aws-cdk/aws-route53resolver';

export interface TgwConnectProps {
  vpc: IVpc;
  transitGatewayId: string;
  tgwDestinationCidr: string;
  resolverRuleId: string;
}

export class TgwConnect extends Construct {
  readonly Vpc: IVpc;
  readonly TgwId: string;
  readonly TgwDestinationCidr: string;
  readonly ResolverRuleId: string;

  constructor(scope: Construct, id: string, props: TgwConnectProps) {
    super(scope, id);

    this.Vpc = props.vpc;
    this.TgwId = props.transitGatewayId;
    this.TgwDestinationCidr = props.tgwDestinationCidr;
    this.ResolverRuleId = props.resolverRuleId;

    const subnetSelect = this.Vpc.selectSubnets();
    for (const subnet of subnetSelect.subnets) {
      new CfnRoute(this, `TGWRoute-${subnet.node.id}`, {
        routeTableId: subnet.routeTable.routeTableId,
        destinationCidrBlock: this.TgwDestinationCidr,
        transitGatewayId: this.TgwId,
      });
    }

    new CfnResolverRuleAssociation(this, 'ResolverRuleAssociation', {
      resolverRuleId: this.ResolverRuleId,
      vpcId: this.Vpc.vpcId,
    });

    new CfnTransitGatewayAttachment(this, 'TransitGatewayAttachment', {
      subnetIds: subnetSelect.subnetIds,
      transitGatewayId: this.TgwId,
      vpcId: this.Vpc.vpcId,
    });
  }
}
