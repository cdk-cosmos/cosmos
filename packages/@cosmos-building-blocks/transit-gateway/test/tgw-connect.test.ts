import { SynthUtils } from '@aws-cdk/assert';
import '@aws-cdk/assert/jest';
import '@aws-cdk/core';
import { TgwConnect } from '../src';
import { App, Stack } from '@aws-cdk/core';
import { Vpc, SubnetType } from '@aws-cdk/aws-ec2';

const app = new App();
const testStack = new Stack(app, 'TestStack');
const myvpc = new Vpc(testStack, 'TestVpc', {
  cidr: '10.180.7.0/24',
  maxAzs: 1,
  subnetConfiguration: [
    {
      name: 'Main',
      subnetType: SubnetType.ISOLATED,
      cidrMask: 26,
    },
    {
      name: 'Redis',
      subnetType: SubnetType.ISOLATED,
      cidrMask: 28,
    },
  ],
});
const transitGatewayId = 'tgw-0d4180d3e7c919164';
const tgwDestinationCidr = '10.0.0.0/8';
const resolverRuleId = 'rslvr-rr-a5c8a7ee5efb41eab';
new TgwConnect(testStack, 'MyTgwConnection', {
  vpc: myvpc,
  transitGatewayId,
  tgwDestinationCidr,
  resolverRuleId,
});
const testStacksynth = SynthUtils.synthesize(testStack);

describe('Transit Gateway Connection', () => {
  test('should create TGW attachments', () => {
    expect(testStacksynth).toHaveResource('AWS::Route53Resolver::ResolverRuleAssociation');
    expect(testStacksynth).toHaveResource('AWS::EC2::TransitGatewayAttachment');
    expect(testStacksynth).toHaveResource('AWS::EC2::Route');
  });

  test('should not create resolver rule association if prop not passed', () => {
    const app1 = new App();
    const testStack1 = new Stack(app1, 'TestStack1');
    const myvpc = new Vpc(testStack1, 'TestVpc', {
      cidr: '10.180.7.0/24',
      maxAzs: 1,
      subnetConfiguration: [
        {
          name: 'Main',
          subnetType: SubnetType.ISOLATED,
          cidrMask: 26,
        },
        {
          name: 'Redis',
          subnetType: SubnetType.ISOLATED,
          cidrMask: 28,
        },
      ],
    });
    new TgwConnect(testStack1, 'MyTgwConnection1', {
      vpc: myvpc,
      transitGatewayId,
      tgwDestinationCidr,
    });
    const testStack1synth = SynthUtils.synthesize(testStack1);
    expect(testStack1synth).not.toHaveResource('AWS::Route53Resolver::ResolverRuleAssociation');
    expect(testStack1synth).toHaveResource('AWS::EC2::TransitGatewayAttachment');
    expect(testStack1synth).toHaveResource('AWS::EC2::Route');
  });

  test('should match snapshot', () => {
    expect(testStacksynth.template).toMatchSnapshot();
  });
});
