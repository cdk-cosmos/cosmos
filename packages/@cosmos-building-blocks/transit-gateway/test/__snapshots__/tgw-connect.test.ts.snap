// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transit Gateway Connection should match snapshot 1`] = `
Object {
  "Resources": Object {
    "MyTgwConnectionResolverRuleAssociation52715D77": Object {
      "Properties": Object {
        "ResolverRuleId": "rslvr-rr-a5c8a7ee5efb41eab",
        "VPCId": Object {
          "Ref": "TestVpcE77CE678",
        },
      },
      "Type": "AWS::Route53Resolver::ResolverRuleAssociation",
    },
    "MyTgwConnectionTGWRouteMainSubnet1D754DB0E": Object {
      "DependsOn": Array [
        "MyTgwConnectionTransitGatewayAttachmentBB73CC5E",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "10.0.0.0/8",
        "RouteTableId": Object {
          "Ref": "TestVpcMainSubnet1RouteTable1567AA0E",
        },
        "TransitGatewayId": "tgw-0d4180d3e7c919164",
      },
      "Type": "AWS::EC2::Route",
    },
    "MyTgwConnectionTGWRouteRedisSubnet117AE6F3C": Object {
      "DependsOn": Array [
        "MyTgwConnectionTransitGatewayAttachmentBB73CC5E",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "10.0.0.0/8",
        "RouteTableId": Object {
          "Ref": "TestVpcRedisSubnet1RouteTableBE392024",
        },
        "TransitGatewayId": "tgw-0d4180d3e7c919164",
      },
      "Type": "AWS::EC2::Route",
    },
    "MyTgwConnectionTransitGatewayAttachmentBB73CC5E": Object {
      "Properties": Object {
        "SubnetIds": Array [
          Object {
            "Ref": "TestVpcMainSubnet1SubnetC0BB40AC",
          },
          Object {
            "Ref": "TestVpcRedisSubnet1Subnet1490AFD4",
          },
        ],
        "TransitGatewayId": "tgw-0d4180d3e7c919164",
        "VpcId": Object {
          "Ref": "TestVpcE77CE678",
        },
      },
      "Type": "AWS::EC2::TransitGatewayAttachment",
    },
    "TestVpcE77CE678": Object {
      "Properties": Object {
        "CidrBlock": "10.180.7.0/24",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/TestVpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "TestVpcMainSubnet1RouteTable1567AA0E": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/TestVpc/MainSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "TestVpcE77CE678",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcMainSubnet1RouteTableAssociation388FC147": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "TestVpcMainSubnet1RouteTable1567AA0E",
        },
        "SubnetId": Object {
          "Ref": "TestVpcMainSubnet1SubnetC0BB40AC",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcMainSubnet1SubnetC0BB40AC": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.180.7.0/26",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/TestVpc/MainSubnet1",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Main",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
        ],
        "VpcId": Object {
          "Ref": "TestVpcE77CE678",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TestVpcRedisSubnet1RouteTableAssociationAFBE0652": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "TestVpcRedisSubnet1RouteTableBE392024",
        },
        "SubnetId": Object {
          "Ref": "TestVpcRedisSubnet1Subnet1490AFD4",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcRedisSubnet1RouteTableBE392024": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/TestVpc/RedisSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "TestVpcE77CE678",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcRedisSubnet1Subnet1490AFD4": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.180.7.64/28",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/TestVpc/RedisSubnet1",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Redis",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
        ],
        "VpcId": Object {
          "Ref": "TestVpcE77CE678",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
  },
}
`;
